Assets/HoloToolkit/Input/Scripts/GestureManipulator.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using HoloToolkit.Unity;
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// A component for moving an object via the GestureManager manipulation gesture.
+    /// </summary>
+    /// <remarks>
+    /// When an active GestureManipulator component is attached to a GameObject it will subscribe
+    /// to GestureManager's manipulation gestures, and move the GameObject when a ManipulationGesture occurs.
+    /// If the GestureManipulator is disabled it will not respond to any manipulation gestures.
+    /// 
+    /// This means that if multiple GestureManipulators are active in a given scene when a manipulation
+    /// gesture is performed, all the relevant GameObjects will be moved.  If the desired behavior is that only
+    /// a single object be moved at a time, it is recommended that objects which should not be moved disable
+    /// their GestureManipulators, then re-enable them when necessary (e.g. the object is focused).
+    /// </remarks>
+    public class GestureManipulator : MonoBehaviour
+    {
+        [Tooltip("How much to scale each axis of hand movement (camera relative) when manipulating the object")]
+        [SerializeField]
+        private Vector3 handPositionScale = new Vector3(2.0f, 2.0f, 4.0f);  // Default tuning values, expected to be modified per application
+
+        private Vector3 initialHandPosition;
+        private Vector3 initialObjectPosition;
+
+        private Interpolator targetInterpolator;
+
+        private bool Manipulating { get; set; }
+
+        private void OnEnable()
+        {
+            if (GestureManager.Instance != null)
+            {
+                GestureManager.Instance.ManipulationStarted += BeginManipulation;
+                GestureManager.Instance.ManipulationCompleted += EndManipulation;
+                GestureManager.Instance.ManipulationCanceled += EndManipulation;
+            }
+            else
+            {
+                Debug.LogError(string.Format("GestureManipulator enabled on {0} could not find GestureManager instance, manipulation will not function", name));
+            }
+        }
+
+        private void OnDisable()
+        {
+            if (GestureManager.Instance)
+            {
+                GestureManager.Instance.ManipulationStarted -= BeginManipulation;
+                GestureManager.Instance.ManipulationCompleted -= EndManipulation;
+                GestureManager.Instance.ManipulationCanceled -= EndManipulation;
+            }
+
+            Manipulating = false;
+        }
+
+        private void BeginManipulation()
+        {
+            if (GestureManager.Instance != null && GestureManager.Instance.ManipulationInProgress)
+            {
+                Manipulating = true;
+
+                targetInterpolator = gameObject.GetComponent<Interpolator>();
```

Comment:
@jbienzms You pretty much nailed it.  The idea here is that if an interpolator is present, we will use it, otherwise we just fall back to setting the transform position, but we don't want to search for it every frame so we just grab it when the manipulation begins.  You could make an argument that we could search for it in update, the cache it for the duration of the manipulation, but I don't know that that is significantly better.  I would prefer to leave this as is unless someone has an alternative proposal that is clearly better.